common/workunit/workunit.cpp

```diff
@@ -1357,9 +1365,16 @@ class GraphScopeIterator : public CInterfaceOf<IConstWUScopeIterator>
             switch (state)
             {
             case SGraph:
-                graphIter->query().getName(StringBufferAdaptor(curScopeName.clear()));
+            {
+                IConstWUGraph & graph = graphIter->query();
+                unsigned wfid = graph.getWfid();
+                curScopeName.clear();
+                if (wfid != 0)
+                    curScopeName.append(WorkflowScopePrefix).append(wfid).append(':');
+                graph.getName(StringBufferAdaptor(curScopeName));
                 scopeType = SSTgraph;
                 return true;
+            }
             case SSubGraph:
                 scopeId.set(SubGraphScopePrefix).append(treeIters.tos().query().getPropInt("@id"));
```

Comment:
shouldn't it prefix the wfid to subgraph scope?

Reply:
no, the wfid is only prefixed onto the graph scope.  it isn't w1:graph2:w1:sg10.